mlcolvar/core/transform/tools/switching_functions.py

```diff
@@ -0,0 +1,81 @@
+import torch
+
+from mlcolvar.core.transform import Transform
+
+
+__all__ = ["SwitchingFunctions"]
+
+class SwitchingFunctions(Transform):
+    """
+    Transform class with some common switching functions
+    """
+    SWITCH_FUNCS = ['Fermi', 'Rational']
+
+    def __init__(self,
+                 in_features : int,
+                 name : str, 
+                 cutoff : float, 
+                 options : dict = None):
+        f"""Initialize switching function object
+
+        Parameters
+        ----------
+        name : str
+            Name of the switching function to be used, available {",".join(self.SWITCH_FUNCS)}
+        cutoff : float
+            Cutoff for the swtiching functions
+        options : dict, optional
+            Dictionary with all the arguments of the switching function, by default None
+        """
+        super().__init__(in_features=in_features, out_features=in_features)
+
+        self.name = name
+        self.cutoff = cutoff
+        if options is None:
+            options = {}
+        self.options = options
+       
+        if name not in self.SWITCH_FUNCS:
+            raise NotImplementedError(f'''The switching function {name} is not implemented in this class. The available options are: {",".join(self.SWITCH_FUNCS)}.
+                    You can initialize it as a method of the SwitchingFunctions class and tell us on Github, contributions are welcome!''')  
+
+    def forward(self, x : torch.Tensor):
+        switch_function = getattr(self, f'{self.name}_switch')
+        y = switch_function(x, self.cutoff, **self.options)
+        return y
+    
+    # ========================== define here switching functions ==========================
+    def Fermi_switch(self,
+                     x : torch.Tensor, 
+                     cutoff : float, 
+                     q : float = 0.01, 
+                     prefactor_cutoff : float = 1.0):
+        y = torch.div( 1, ( 1 + torch.exp( torch.div((x - prefactor_cutoff*cutoff) , q ))))
+        return y
+
+    def Rational_switch(self,
```

Comment:
Would it be a good idea to implement these as two separate classes with a common base class instead?

Reply:
If you mean so that you can pass a simple string to decide which one to use, yeah, it make a lot of sense. You can still do it using an object-oriented design using a factory-like pattern if you wanted. For just two classes, something like this could be enough:
```python
def get_switching_function(type_str):
    if type_str == 'Fermi':
        return FermiSwitchingFunction
    elif ...:
        ...
    else:
        raise ValueError
```
and then in the classes that need the switching function you just do
```python
get_switching_function('Fermi')(in_features, cutoff, options)
```
Factory patterns are often implemented with a registry rather than hardcoding with ``if-else``s, but for now I'd keep it as simple as possible.

Again, this is matter of taste so it's not essential. The current implementation works.

Reply:
I already gave you the spaces before the double dots. Jokes apart, I also kept a structure similar to the way we pass the options to the CV models. So maybe we can keep it